plugins/cursecheck.cpp

```diff
@@ -157,7 +157,7 @@ command_result cursecheck (color_ostream &out, vector <string> & parameters)
     }
 
     // check whole map if no unit is selected
-    for(df::unit *unit : selected_unit ? vector{ selected_unit } : world->units.all)
+    for(df::unit *unit : selected_unit[0] ? selected_unit : world->units.all)
```

Comment:
my first version did just that, but it looked messy (because of the required null check), but maybe I can clean it up more

Reply:
#5150 is arguably cleaner

Reply:
while i'm not going to change the code we have now, this would arguably have been a good use for a `std::span`

Reply:
for my own education, how would that have worked? a ternary expression must have the same type in both of its branches. How would we wrap `units.all` to have the same type as the inline span?

Reply:
i was thinking that we knew where in the `all` vector the selected unit is, but we don't, so there's no value in having either the entire units vector or just part of it, which is what `std::span` would have done for us here

we want here an object that holds a reference to a vector which is either owning or nonowning, and the object knows which. this is because in the single-unit case, we need a temporary vector which will be destructed when we're done with it, but in the all-units case we do not want to destruct the vector to be iterated. there are ways to do this with `shared_ptr` but they're not less oblique than the solution used